pkg/chaincode/querycommitted.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package chaincode
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hyperledger/fabric-admin-sdk/pkg/identity"
+	"github.com/hyperledger/fabric-admin-sdk/pkg/internal/proposal"
+
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer/lifecycle"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+// QueryCommitted chaincode on a specific peer.
+func QueryCommitted(ctx context.Context, connection grpc.ClientConnInterface, signingID identity.SigningIdentity, channelID string) (*lifecycle.QueryChaincodeDefinitionsResult, error) {
+	queryArgs := &lifecycle.QueryChaincodeDefinitionsArgs{}
```

Comment:
https://github.com/hyperledger/fabric/blob/main/internal/peer/lifecycle/chaincode/querycommitted.go#L223-L227
if I am correct, we can add chaincode name a parameter? ref to fabric code as example.

Reply:
wait, @bestbeforetoday , we removed name as parameter at https://github.com/hyperledger/fabric-protos-go-apiv2/blob/main/peer/lifecycle/lifecycle.pb.go#L1290-L1328 ?
are we able to pass QueryChaincodeDefinitionsArgs with name in fabric-protos-go-apiv2?

Reply:
wait, @bestbeforetoday , we removed name as parameter at https://github.com/hyperledger/fabric-protos-go-apiv2/blob/main/peer/lifecycle/lifecycle.pb.go#L1290-L1328 ?
are we able to pass QueryChaincodeDefinitionsArgs with name in fabric-protos-go-apiv2?

Reply:
There are currently two options for querying committed chaincodes in Fabric. One of them queries all committed in the channel and the other filters it by chaincode name.

One uses `QueryChaincodeDefinitions` tx name and the other uses `QueryChaincodeDefinition`.

Should I proceed to add both options then? In separate functions or the same?

Reply:
I suppose if possible, we should support both query all and query by name.

Reply:
I'll let this PR be merged then, and make another adding support to query by chaincode name.

Reply:
> are we able to pass QueryChaincodeDefinitionsArgs with name in fabric-protos-go-apiv2?

We can use `QueryChaincodeDefinitionArgs` with name. I can do this after this PR.

```type QueryChaincodeDefinitionArgs struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	Name string `protobuf:"bytes,1,opt,name=name,proto3" json:"name,omitempty"`
}```

Reply:
by default, I will merge this PR before Monday, will keep this PR open for other reviewers. :-)
it's 11 pm my local time, I am going to have a rest.

Reply:
@samuelvenzi , you are able to have another PR for query by name after this PR been merged.

Reply:
I personally prefer `QueryCommitted` and `QueryCommittedWithName`. I didn't realize the information was slightly different, but you're right.